casper/src/main/scala/coop/rchain/casper/merging/DeployChainIndex.scala

```diff
@@ -12,6 +13,7 @@ final case class DeployIdWithCost(id: ByteString, cost: Long)
 
 /** index of deploys depending on each other inside a single block (state transition) */
 final case class DeployChainIndex(
+    blockHash: BlockHash, // user might submit the same deploy to multiple validators, so index should include block hash as well
```

Comment:
This is added in case the same deploy is put in multiple sibling blocks. I'm still not sure this is a good idea - DeployChainIndex is part of block index, so its duplication.

Reply:
Also this makes me thinking: now we put rejected deploys in the block, but there might be multiple insertions of the same rejected deploy in the case I'm talking about. Is that OK or we should add data about block hash from where deploy has been rejected?

Reply:
On which stage are now detected duplicated deploys? They can influence conflict detection. I'm not sure when is the best time to remove duplicates.

Reply:
This is good question.
https://github.com/rchain/rchain/blob/5d94c2d976f61e897634f3d8bb00f6bc29588d0e/casper/src/main/scala/coop/rchain/casper/util/rholang/CasperRhoRuntimeSyntax.scala#L378-L385

Unforgeable names, so channels, so produces and consumes in event log depend on deployer public key and deploy timestamp.

I think we can expand notion of conflict to add `contains the same event`, but then random seed have to not include validator (as we were discussing iirc) or anything that is outside of deployer control.